### PR TITLE
Support setting new URLSessionConfiguration

### PR DESCRIPTION
**Pull request by angelix**
Add support to update the URLSessionConfiguration

### CODE REVIEW

**Comment by ArtursKadikis** [comment]
Hello, what would be the purpose for this?

**Comment by angelix** [comment]
> Hello, what would be the purpose for this?

To set/update network proxy settings.

**Comment by ArtursKadikis** [comment]
what kind of proxy setting would you be setting? could you mention the specific fields?

**Comment by angelix** [comment]
> what kind of proxy setting would you be setting? could you mention the specific fields?

Here is our implementation using this feature.
https://github.com/Blockstream/green_ios/blob/5d3b31cbe794dc4b03454484eb98042b1a02d5fc/gaios/Services/AnalyticsManager/AnalyticsManager.swift#L145-L194

We use it as a http proxy or socks5 for tor communication.

File: CountlyConnectionManager.m [hunk 1]
```diff
@@ -109,6 +109,20 @@ - (void)setHost:(NSString *)host
     }
 }
 
+- (void)setURLSessionConfiguration:(NSURLSessionConfiguration *)URLSessionConfiguration
+{
+    if (URLSessionConfiguration == nil)
```
[the remark this answers is not in the document]
Comment (angelix) [reply CountlyConnectionManager.m]: Fixed in ac3ca53a471c458a9bf266aed3e762a4b36f7c19

**Review by erkanyildiz — APPROVED** [review]

**Pull request merged**